### PR TITLE
Sample weight sliced to work with cross validation, issue #358

### PR DESCRIPTION
**Pull request by rg2410**
Changes that could fix situation described in issue #358

### CODE REVIEW

**Comment by rg2410** [comment]
Thanks for the suggestions @lopuhin, I've added them in the two last commits

File: eli5/sklearn/permutation_importance.py [hunk 1]
```diff
@@ -214,8 +214,13 @@ def _cv_scores_importances(self, X, y, groups=None, **fit_params):
...
         feature_importances = []  # type: List
         base_scores = []  # type: List[float]
+        weights = fit_params.get('sample_weight', None)
+        fit_params.pop('sample_weight', None)
         for train, test in cv.split(X, y, groups):
-            est = clone(self.estimator).fit(X[train], y[train], **fit_params)
+            if weights is None:
+                est = clone(self.estimator).fit(X[train], y[train], **fit_params)
+            else:
+                est = clone(self.estimator).fit(X[train], y[train], sample_weight=weights[train], **fit_params)
```
Comment (lopuhin) [inline eli5/sklearn/permutation_importance.py]: This does the right thing, but it's possible to improve this a bit - instead of repeating almost the same line twice, it's possible to modify ``fit_params`` or create new ``fold_fit_params``, putting modified ``sample_weight`` there - then the same call can be done for both cases, avoiding repetition.

File: eli5/sklearn/permutation_importance.py [hunk 2]
```diff
@@ -214,8 +214,13 @@ def _cv_scores_importances(self, X, y, groups=None, **fit_params):
         cv = check_cv(self.cv, y, is_classifier(self.estimator))
         feature_importances = []  # type: List
         base_scores = []  # type: List[float]
+        weights = fit_params.get('sample_weight', None)
+        fit_params.pop('sample_weight', None)
```
Comment (lopuhin) [inline eli5/sklearn/permutation_importance.py]: An minor improvement is possible here: ``pop`` already returns what it popped, so these two lines could be reduced to one:
```
 weights = fit_params.pop('sample_weight', None)
```

Comment (rg2410) [reply eli5/sklearn/permutation_importance.py]: Done

File: eli5/sklearn/permutation_importance.py (hunk 1 above)
Comment (rg2410) [reply eli5/sklearn/permutation_importance.py]: have a look at the new changes and see if they make sense. Good suggestions
